### PR TITLE
Mention both release *and* edition breakage for never type lints

### LINKED ISSUE #132930: 2024 `dependency_on_unit_never_type_fallback` doesn't display during 2024 edition migration
I'm not entirely certain what is wrong with this particular code, but it fails in the 2024 edition without generating any warnings.

screenshots@0.8.10
https://crater-reports.s3.amazonaws.com/pr-132712/try%23da25749bf5e6ba5ed862ff361c19afff2a986b2d/reg/screenshots-0.8.10/log.txt

The error after updating to 2024 is:

```
error[E0277]: the trait bound `!: ReadAll` is not satisfied
   --> src/linux/wayland_screenshot.rs:70:11
    |
70  |     proxy.method_call(
    |           ^^^^^^^^^^^ the trait `ReadAll` is not implemented for `!`
    |
    = help: the following other types implement trait `ReadAll`:
              ()
              (A, B)
              (A, B, C)
              (A, B, C, D)
              (A, B, C, D, E)
              (A, B, C, D, E, F)
              (A, B, C, D, E, F, G)
              (A, B, C, D, E, F, G, H)
            and 28 others
    = note: this error might have been caused by changes to Rust's type-inference algorithm (see issue #48950 <https://github.com/rust-lang/rust/issues/48950> for more information)
    = help: did you intend to use the type `()` here instead?
note: required by a bound in `Proxy::<'a, C>::method_call`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/dbus-0.9.7/src/blocking.rs:396:35
    |
396 |     pub fn method_call<'i, 'm, R: ReadAll, A: AppendAll, I: Into<Interface<'i>>, M: Into<Member<'m>>>(&self, i: I, m: M, args: A) -> Result<R, Error> {
    |                                   ^^^^^^^ required by this bound in `Proxy::<'a, C>::method_call`

error[E0277]: the trait bound `!: ReadAll` is not satisfied
   --> src/linux/wayland_screenshot.rs:126:11
    |
126 |     proxy.method_call(
    |           ^^^^^^^^^^^ the trait `ReadAll` is not implemented for `!`
    |
    = help: the following other types implement trait `ReadAll`:
              ()
              (A, B)
              (A, B, C)
              (A, B, C, D)
              (A, B, C, D, E)
              (A, B, C, D, E, F)
              (A, B, C, D, E, F, G)
              (A, B, C, D, E, F, G, H)
            and 28 others
    = note: this error might have been caused by changes to Rust's type-inference algorithm (see issue #48950 <https://github.com/rust-lang/rust/issues/48950> for more information)
    = help: did you intend to use the type `()` here instead?
note: required by a bound in `Proxy::<'a, C>::method_call`
   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/dbus-0.9.7/src/blocking.rs:396:35
    |
396 |     pub fn method_call<'i, 'm, R: ReadAll, A: AppendAll, I: Into<Interface<'i>>, M: Into<Member<'m>>>(&self, i: I, m: M, args: A) -> Result<R, Error> {
    |                                   ^^^^^^^ required by this bound in `Proxy::<'a, C>::method_call`
```

The relevant types are from the dbus crate.

I would expect that if this is related to [`dependency_on_unit_never_type_fallback`](https://doc.rust-lang.org/nightly/rustc/lints/listing/warn-by-default.html#dependency-on-unit-never-type-fallback) that it should generate a warning before migrating to 2024.

### Meta

`rustc --version --verbose`:
```
rustc 1.84.0-nightly (143ce0920 2024-11-10)
binary: rustc
commit-hash: 143ce0920a2307b19831160a01f06f107610f1b2
commit-date: 2024-11-10
host: aarch64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.3
```

### PR DESCRIPTION
**Pull request by WaffleLapkin**
This PR makes ~~two changes~~ a change to the never type lints (`dependency_on_unit_never_type_fallback` and `never_type_fallback_flowing_into_unsafe`):
1.  Change the wording of the note to mention that the breaking change will be made in an edition _and_ in a future release
2. ~~Make these warnings be reported in deps (hopefully the lints are matured enough)~~

r? @compiler-errors
cc @ehuss 
closes #132930

### CODE REVIEW

**Review by compiler-errors — APPROVED** [review]
ty

**Comment by compiler-errors** [comment]
I actually think that we should not be warning this in deps yet... warning this in deps is currently unactionable from the user perspective, and I think it may need additional approval from T-lang 🤔

**Comment by WaffleLapkin** [comment]
@compiler-errors I've reverted the change making the lints reported in deps.

**Comment by compiler-errors** [comment]
ty :>

File: compiler/rustc_lint_defs/src/builtin.rs [hunk 1]
```diff
@@ -4185,7 +4185,7 @@ declare_lint! {
     Warn,
     "never type fallback affecting unsafe function calls",
     @future_incompatible = FutureIncompatibleInfo {
-        reason: FutureIncompatibilityReason::FutureReleaseSemanticsChange,
+        reason: FutureIncompatibilityReason::EditionAndFutureReleaseError(Edition::Edition2024),
```
Comment (compiler-errors) [inline compiler/rustc_lint_defs/src/builtin.rs:4188]: hm, do we want to maintain the fact that this is a semantics change? i think we kinda care

Comment (compiler-errors) [reply compiler/rustc_lint_defs/src/builtin.rs:4188]: whoops i forgot to finish my sentece -- we kinda care about this difference, right?

Comment (WaffleLapkin) [reply compiler/rustc_lint_defs/src/builtin.rs:4188]: Yes, we should maintain this fact. Oops ^^'

Comment (WaffleLapkin) [reply compiler/rustc_lint_defs/src/builtin.rs]: Fixed.

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**